File: src/src/org/github/olloginov/ideataskbrowser/view/TaskTreeNode.java
package org.github.olloginov.ideataskbrowser.view;

import com.intellij.tasks.Task;
import com.intellij.ui.ColoredTextContainer;
import com.intellij.ui.SimpleTextAttributes;

import javax.swing.*;
import javax.swing.tree.DefaultMutableTreeNode;

public class TaskTreeNode extends DefaultMutableTreeNode implements CustomIcon, CustomLabel {
    public TaskTreeNode(Task task) {
        super(task);
    }

    public Task getTask() {
        return (Task) getUserObject();
    }

    @Override
    public Icon getIcon() {
        return getTask().getIcon();
    }

    @Override
    public String toString() {
        return getTask().getPresentableName();
    }

    public void setTask(Task task) {
        setUserObject(task);
    }

    @Override
    public void setLabel(ColoredTextContainer coloredTextContainer) {
        coloredTextContainer.append(getTask().toString(), SimpleTextAttributes.REGULAR_ATTRIBUTES);
    }
}

File: src/main/kotlin/org/github/olloginov/ideataskbrowser/view/TaskTreeNode.kt
package org.github.olloginov.ideataskbrowser.view

import com.intellij.tasks.Task
import com.intellij.ui.ColoredTextContainer
import com.intellij.ui.SimpleTextAttributes
import javax.swing.Icon
import javax.swing.tree.DefaultMutableTreeNode

class TaskTreeNode(task: Task) : DefaultMutableTreeNode(task), CustomIcon, CustomLabel {
	fun getTask(): Task = getUserObject() as Task

	override fun getIcon(): Icon = getTask().icon

	override fun toString(): String = getTask().presentableName

	fun setTask(task: Task) {
		setUserObject(task)
	}

	override fun setLabel(coloredTextContainer: ColoredTextContainer) {
		coloredTextContainer.append(getTask().toString(), SimpleTextAttributes.REGULAR_ATTRIBUTES)
	}
}
